src/core/server/integration_tests/saved_objects/migrations/zdt_2/sor_higher_version_docs.test.ts

```diff
@@ -8,57 +8,49 @@
 
 import { pick, range } from 'lodash';
 import Path from 'path';
-import fs from 'fs/promises';
-import { type TestElasticsearchUtils } from '@kbn/core-test-helpers-kbn-server';
 import '../jest_matchers';
 import { ISavedObjectsRepository } from '@kbn/core-saved-objects-api-server';
-import { SavedObjectsModelVersionMap } from '@kbn/core-saved-objects-server';
-import { getKibanaMigratorTestKit, startElasticsearch } from '../kibana_migrator_test_kit';
-import { delay, createType } from '../test_utils';
-import { getBaseMigratorParams } from '../fixtures/zdt_base.fixtures';
+import { createType } from '../test_utils';
 import { SavedObjectsBulkCreateObject } from '@kbn/core-saved-objects-api-server';
+import { createModelVersionTestBed } from '@kbn/core-test-helpers-model-versions';
 
 export const logFilePath = Path.join(__dirname, 'sor_higher.test.log');
 
+const modelVersionTestBed = createModelVersionTestBed();
+
 describe('Higher version doc conversion', () => {
-  let esServer: TestElasticsearchUtils['es'];
   let repositoryV1: ISavedObjectsRepository;
   let repositoryV2: ISavedObjectsRepository;
 
-  const getTestType = ({ includeVersion2 }: { includeVersion2: boolean }) => {
-    const modelVersions: SavedObjectsModelVersionMap = {
-      1: {
-        changes: [],
-        schemas: {
-          forwardCompatibility: (attrs: any) => {
-            return pick(attrs, 'text', 'bool');
+  const getTestType = () => {
```

Comment:
I did the exercise of rewriting this test (that was doing exactly what this testkit was meant for) using the new tooling, and I'm quite happy, as the code feels more readable and maintainable now.